### ISSUE TITLE
bug with break + babel inserted iife

### ISSUE THREAD

**Issue by monsanto** [babel/babel#998]
```
for (let i = 0; i < 3; i++) {
    () => { i }
    console.log(i)
    break;
    continue;
}
```

this should just print `0`, but babel will print `0 1 2`. This is because:

```
for (var i = 0; i < 3; i++) {
    var _ret = (function (i) {
        (function () {
            i;
        });
        console.log(i);
        return "break";
        return "continue";
    })(i);

    switch (_ret) {
        case "break":
            break;

        case "continue":
            continue;}
}
```

a switch is used, which catches the break intended for the loop. Should either change to if statements or do a labelled break.

**Comment by sebmck** [comment]
Incredibly sorry for the delay, fixed as of 4.7.9.

**Comment by monsanto** [comment]
No worries @sebmck, thanks for your hard work.
